lib/services/local/access.go

```diff
@@ -219,6 +222,63 @@ func (s *AccessService) DeleteAllLocks(ctx context.Context) error {
 	return s.DeleteRange(ctx, startKey, backend.RangeEnd(startKey))
 }
 
+// ReplaceRemoteLocks replaces the set of locks associated with a remote cluster.
+func (s *AccessService) ReplaceRemoteLocks(ctx context.Context, clusterName string, newRemoteLocks []types.Lock) error {
+	s.remoteLocksMutex.Lock()
+	defer s.remoteLocksMutex.Unlock()
```

Comment:
The mutex serves to emulate a poor man's atomic transaction. Although this isn't achieved on the backend level (in the sense that a failed backend operation rollbacks the already performed operations), we at least prevent interference from another `ReplaceRemoteLocks` call.

I don't understand your point about the load balancer scenario. Each auth server would keep its own mutex so this shouldn't impact simultaneous `ReplaceRemoteLocks` among multiple auth servers at all...?

Reply:
> We at least prevent interference from another ReplaceRemoteLocks call

I meant scenario where ReplaceRemoteLocks function is called at the same time by two difference clients and requests can reaches two independent auth services connected to the same backend. In that case the final result is not deterministic and depends on Get/Update/Put order invoked by two different auth services. So just wonder if [acquiring backend lock ](https://github.com/gravitational/teleport/blob/f86068e0d8f6ff4a8b1cb5bd8d7b72df41c55e36/lib/backend/helpers.go#L128) usage in not needed here to prevent such situation.

Reply:
@smallinsky Good point. Rewritten using `backend.RunWhileLocked`.